src/app/settings/admin/constraint-templates/component.ts

```diff
@@ -0,0 +1,139 @@
+    private readonly _matDialog: MatDialog
+  ) {}
+
+  ngOnInit() {
+    this.dataSource.data = this.constraintTemplates;
+    this.dataSource.sort = this.sort;
+    this.dataSource.paginator = this.paginator;
+    this.sort.active = 'templateName';
+    this.sort.direction = 'asc';
+
+    this._opaService.constraintTemplates.pipe(takeUntil(this._unsubscribe)).subscribe(constraintTemplates => {
+      this.constraintTemplates = constraintTemplates;
+      this.dataSource.data = this.constraintTemplates;
+    });
+
+    this._userService.currentUserSettings.pipe(takeUntil(this._unsubscribe)).subscribe(settings => {
+      this.paginator.pageSize = settings.itemsPerPage;
+      this.dataSource.paginator = this.paginator; // Force refresh.
+    });
+  }
+
+  ngOnChanges(): void {
+    this.dataSource.data = this.constraintTemplates;
+  }
+
+  ngOnDestroy(): void {
+    this._unsubscribe.next();
+    this._unsubscribe.complete();
+  }
+
+  isPaginatorVisible(): boolean {
+    return (
+      this.constraintTemplates &&
+      this.constraintTemplates.length > 0 &&
+      this.paginator &&
+      this.constraintTemplates.length > this.paginator.pageSize
+    );
+  }
+
+  add(): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Add Constraint Template',
+        mode: Mode.Add,
+        confirmLabel: 'Add',
+      },
+    };
+
+    this._matDialog
+      .open(ConstraintTemplateDialog, dialogConfig)
+      .afterClosed()
+      .pipe(take(1))
+      .subscribe(_ => {});
+  }
+
+  edit(constraintTemplate: ConstraintTemplate): void {
+    const dialogConfig: MatDialogConfig = {
+      data: {
+        title: 'Edit Constraint Template',
+        constraintTemplate: constraintTemplate,
+        mode: Mode.Edit,
+        confirmLabel: 'Edit',
+      },
+    };
+
+    this._matDialog
+      .open(ConstraintTemplateDialog, dialogConfig)
+      .afterClosed()
+      .pipe(take(1))
+      .subscribe(_ => {});
+  }
+
+  delete(constraintTemplate: ConstraintTemplate): void {
+    const dialogConfig: MatDialogConfig = {
+      disableClose: false,
+      hasBackdrop: true,
+      data: {
+        title: 'Delete Constraint Template',
+        message: `Are you sure you want to delete the constraint template ${constraintTemplate.name}?`,
+        confirmLabel: 'Delete',
+        warning: 'Deleting this constraint template will cause all constraints related to it to be deleted as well.',
+      },
+    };
+
+    this._matDialog
+      .open(ConfirmationDialogComponent, dialogConfig)
+      .afterClosed()
+      .pipe(filter(isConfirmed => isConfirmed))
+      .pipe(switchMap(_ => this._opaService.deleteConstraintTemplate(constraintTemplate.name)))
+      .pipe(take(1))
+      .subscribe(_ => {
+        this._notificationService.success(`The constraint template ${constraintTemplate.name} was deleted`);
+        setTimeout(() => this._opaService.refreshConstraintTemplates(), this._defaultTimeout);
```

Comment:
I'd remove this timeout. There is now a good 10s forced refresh so even if API doesn't remove it in time and this refresh won't get it then it will be refreshed after few seconds.

Reply:
good catch 👍 removed.